syntax = "proto2";

package e2e_noa;

import "affine_transformation.proto";

enum ObjectRoadStatus {
  ORS_NONE = 0;
  ORS_OFF_ROAD = 1;
  ORS_ON_ROAD = 2;
}

enum ObjectIntersectionStatus {
  OIS_NONE = 0;
  OIS_OUT_INTERSECTION = 1;
  OIS_IN_INTERSECTION = 2;
}

message PredictedTrajectoryPointProto {
  // Spatial quantities.
  optional Vec2dProto pos = 1;
  optional double s = 2;      // Arc length (starting from first point).
  optional double theta = 3;  // Facing orientation.
  optional double kappa = 4;  // dtheta/ds.

  // Temporal quantities.
  optional double t = 5;  // Time (relative to first point).
  optional double v = 6;  // Speed. Can not be negative.
  optional double a = 7;  // dv/dt.
}
